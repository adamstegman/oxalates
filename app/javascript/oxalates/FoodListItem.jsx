import React from 'react';
import PropTypes from 'prop-types';

import { cssify } from './cssify';
import { foodPropType } from './foodPropType';
import { listPropType } from './listPropType';

import './FoodListItem.scss';

export class FoodListItem extends React.Component {
  render() {
    const {
      food,
      list,
    } = this.props;

    const listClassName = cssify(list.name);
    return (
      <li className={`food-list-item ${listClassName}`}>
        {food.name} - {food.oxalateMg.toFixed(2)}mg - {food.serving}
      </li>
    );
  }
};

FoodListItem.propTypes = {
  food: foodPropType.isRequired,
  list: listPropType.isRequired,
};
