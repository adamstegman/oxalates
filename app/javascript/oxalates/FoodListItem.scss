@import "./lists";

$food_list_item_padding: 0.75em;
$food_list_item_font_size: 16px;
$destroy_food_list_item_icon_height: 22px;
$destroy_food_list_item_icon_width: 22px;
$destroy_food_list_item_link_height: 44px;
$destroy_food_list_item_link_width: 44px;

.food-list-item {
  padding: $food_list_item_padding;

  font-size: $food_list_item_font_size;
}

.destroy-food-list-item {
  width: $destroy_food_list_item_icon_width;
  height: $destroy_food_list_item_icon_height;
  padding: #{($destroy_food_list_item_link_height - $destroy_food_list_item_icon_height) / 2}
           #{($destroy_food_list_item_link_width  - $destroy_food_list_item_icon_width)  / 2};
}

.edit-food-list-item {
  @extend .food-list-item;

  display: block;
  -webkit-flex-grow: 2;
  flex-grow: 2;

  text-decoration: none;
}

.food-list-item-name {
  display: none;
}
input.food-list-item-name {
  @extend .food-list-item;

  width: 100%;

  -webkit-appearance: none;
}
