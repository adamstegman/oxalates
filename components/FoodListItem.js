import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import Colors from '../constants/Colors';

export class FoodListItem extends React.Component {
  render() {
    const food = this.props.food;
    const oxalates = food.oxalate_mg.toFixed(2);
    const nameStyle = StyleSheet.flatten([styles.text]);
    const infoContainerStyle = StyleSheet.flatten([styles.container, styles.infoContainer]);
    const infoTextStyle = StyleSheet.flatten([nameStyle, styles.infoText]);
    return(
      <View style={styles.container}>
        <Text numberOfLines={1} style={nameStyle}>{food.name}</Text>
        <View style={infoContainerStyle}>
          <Text numberOfLines={1} style={infoTextStyle}>{oxalates}mg</Text>
          <Text numberOfLines={1} style={infoTextStyle}>{food.serving}</Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 4,
    marginLeft: 12,
    marginBottom: 4,
    marginRight: 12,
  },
  text: {
    flex: 7,
    fontSize: 16,
    lineHeight: 40,
  },
  infoContainer: {
    flex: 3,
    flexDirection: 'column',
  },
  infoText: {
    color: Colors.lightText,
    fontSize: 12,
    lineHeight: 16,
    textAlign: 'right',
  },
});
