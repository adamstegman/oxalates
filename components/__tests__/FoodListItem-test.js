import 'react-native';
import React from 'react';
import { FoodListItem } from '../FoodListItem';
import renderer from 'react-test-renderer';

it('renders a food as a list item', () => {
  const tree = renderer.create(<FoodListItem food={{name: 'Test', serving: '1 cup', oxalate_mg: 30}} />).toJSON();

  expect(tree).toMatchSnapshot();
});
