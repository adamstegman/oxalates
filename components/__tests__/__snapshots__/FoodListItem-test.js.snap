// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a food as a list item 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 4,
      "marginLeft": 12,
      "marginRight": 12,
      "marginTop": 4,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "#e00",
        "flex": 7,
        "fontSize": 16,
        "lineHeight": 40,
      }
    }
  >
    Test
  </Text>
  <View
    style={
      Object {
        "flex": 3,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 4,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#666",
          "flex": 7,
          "fontSize": 12,
          "lineHeight": 16,
          "textAlign": "right",
        }
      }
    >
      30.00
      mg
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#666",
          "flex": 7,
          "fontSize": 12,
          "lineHeight": 16,
          "textAlign": "right",
        }
      }
    >
      1 cup
    </Text>
  </View>
</View>
`;
